src/SablierV2MerkleLockupLT.sol

```diff
@@ -0,0 +1,171 @@
+
+        // Effects: mark the index as claimed.
+        _claimedBitMap.set(index);
+
+        // Interactions: create the stream via {SablierV2LockupTranched}.
+        streamId = LOCKUP_TRANCHED.createWithDurations(
+            LockupTranched.CreateWithDurations({
+                sender: admin,
+                recipient: recipient,
+                totalAmount: amount,
+                asset: ASSET,
+                cancelable: CANCELABLE,
+                transferable: TRANSFERABLE,
+                tranches: tranches,
+                broker: Broker({ account: address(0), fee: ud(0) })
+            })
+        );
+
+        // Log the claim.
+        emit Claim(index, recipient, amount, streamId);
+    }
+
+    /// @dev Calculates the stream tranches based on Merkle tree amount and predefined percentage for each tranche.
+    function _calculateTranches(uint128 amount)
+        internal
+        view
+        returns (LockupTranched.TrancheWithDuration[] memory tranches)
+    {
+        // Load the tranches in memory to save gas.
+        MerkleLockupLT.TrancheWithPercentage[] memory tranchesWithPercentages = _tranchesWithPercentages;
+
+        // Declare the variables need for calculation.
+        UD60x18 trancheAmountsSum;
+        uint256 trancheCount = tranchesWithPercentages.length;
+        tranches = new LockupTranched.TrancheWithDuration[](trancheCount);
+
+        UD60x18 udAmount = ud(amount);
+
+        // Iterate over each tranche to calculate its amount based on its percentage.
+        for (uint256 i = 0; i < trancheCount; ++i) {
+            // Convert the tranche's percentage to `UD60x18` for calculation.
+            UD60x18 percentage = (tranchesWithPercentages[i].amountPercentage).intoUD60x18();
+
+            // Calculate the tranche's amount by applying its percentage to the `amount`.
+            UD60x18 trancheAmount = udAmount.mul(percentage);
+
+            // Sum all tranche amounts.
+            trancheAmountsSum = trancheAmountsSum.add(trancheAmount);
+
+            // Assign calculated amount and duration to the tranche.
+            tranches[i] = LockupTranched.TrancheWithDuration({
+                amount: trancheAmount.intoUint128(),
+                duration: tranchesWithPercentages[i].duration
+            });
+        }
+
+        // Safe Interactions: query the protocol fee. This is safe because it's a known Sablier contract that does
+        // not call other unknown contracts.
+        UD60x18 protocolFee = LOCKUP_TRANCHED.comptroller().protocolFees(ASSET);
+
+        // Adjust the last tranche amount if the protocol fee is greater than zero.
+        // This is needed to ensure the protocol invariat: sum of tranche amounts equals the deposit amount.
+        if (protocolFee.gt(ud(0))) {
+            UD60x18 protocolFeeAmount = udAmount.mul(protocolFee);
+            tranches[trancheCount - 1].amount -= protocolFeeAmount.intoUint128();
```

Comment:
This works only when `tranches[trancheCount - 1].amount` > `protocolFeeAmount`. Otherwise the correct pseudo code to implement it would be

```solidity
if (tranches[trancheCount - 1].amount > protocolFeeAmount.intoUint128()) {
    tranches[trancheCount - 1].amount -= protocolFeeAmount.intoUint128();
} else if (tranches[trancheCount - 2].amount > protocolFeeAmount.intoUint128() - tranches[trancheCount - 1].amount) {
    tranches[trancheCount - 2].amount -= (protocolFeeAmount.intoUint128() - tranches[trancheCount - 1].amount);
} else if 
.......
```

Reply:
correct, so the better version would be to:
- calculate deposit amount: `totalAmount - protocolFeeAmount`
- calculate tranches amounts from `depositAmount` instead of `totalAmount`

Reply:
in the end, we are going to remove the fees, so this handle is not going to be necessary